db/migrate/20240213151647_create_mentor_memberships.rb

```diff
@@ -0,0 +1,11 @@
+class CreateMentorMemberships < ActiveRecord::Migration[7.1]
+  def change
+    create_table :mentor_memberships, id: :uuid do |t|
+      t.string :type
+      t.references :mentor, null: false, foreign_key: true, type: :uuid
+      t.references :school, null: false, foreign_key: true, type: :uuid
```

Comment:
I think we should include some database indexes in here, too.

What about a composite index like this:

```ruby
t.index %i[type school_id mentor_id], unique: true
```

This should support the read patterns:

- Find all mentors associated with a school (using a partial index)
- Enforce unique membership constraint (using the full 'unique' index)

And then another index for mentors scoped to a specific service:

```ruby
t.index %i[type mentor_id]
```

Which would support the read pattern:

- Find all schools associated with a mentor

Reply:
c06a141d8659b694ad202fd79569d29eaeceaf2c